contracts/extensions/VotingBase.sol

```diff
@@ -0,0 +1,1108 @@
+
+    fractionUserInfluence = fractionUserInfluence / influence.length;
+
+    uint256 totalStake = add(motion.stakes[YAY], motion.stakes[NAY]);
+    return wmul(wmul(fractionUserInfluence, totalStake), voterRewardFraction);
+  }
+
+  /// @notice Get the range of potential rewards for a voter on a specific motion, intended to be
+  /// used when the motion is in the reveal state.
+  /// Once a motion is in the reveal state the reward is known, and getVoterRewardRange should be used.
+  /// @param _motionId The id of the motion
+  /// @param _user The address of the user
+  /// @return The voter reward
+  function getVoterRewardRange(uint256 _motionId, address _user) public view returns (uint256, uint256) {
+    Motion storage motion = motions[_motionId];
+
+    uint256[] memory influence = getInfluence(_motionId, _user);
+    assert(influence.length == motion.totalVotes.length);
+
+    uint256 minFractionUserInfluence;
+    uint256 maxFractionUserInfluence;
+
+    // The minimum reward is when everyone has voted, with a total weight of motion.maxVotes
+    // The maximum reward is when this user is the only other person who votes (if they haven't already),
+    //  aside from those who have already done so
+
+    for (uint256 i; i < influence.length; i++) {
+      // If user hasn't voted, add their influence to totalVotes
+      uint256 pendingVote = (voteSecrets[_motionId][_user] == bytes32(0)) ? influence[i] : 0;
+
+      // TODO: Divide-by-zero ?
+      minFractionUserInfluence = add(minFractionUserInfluence, wdiv(influence[i], motion.maxVotes[i]));
+      maxFractionUserInfluence = add(maxFractionUserInfluence, wdiv(influence[i], add(motion.totalVotes[i], pendingVote)));
+    }
+
+    minFractionUserInfluence = minFractionUserInfluence / influence.length;
+    maxFractionUserInfluence = maxFractionUserInfluence / influence.length;
+    uint256 totalStake = add(motion.stakes[YAY], motion.stakes[NAY]);
+
+    return (
+      wmul(wmul(minFractionUserInfluence, totalStake), voterRewardFraction),
+      wmul(wmul(maxFractionUserInfluence, totalStake), voterRewardFraction)
+    );
+  }
+
+  /// @notice Get the staker reward
+  /// @param _motionId The id of the motion
+  /// @param _staker The staker's address
+  /// @param _vote The vote (0 = NAY, 1 = YAY)
+  /// @return The staker reward and the reputation penalty (if any)
+  function getStakerReward(
+    uint256 _motionId,
+    address _staker,
+    uint256 _vote
+  )
+    public
+    view
+    returns (uint256, uint256)
+  {
+    Motion storage motion = motions[_motionId];
+
+    uint256 totalSideStake = add(motion.stakes[_vote], motion.pastVoterComp[_vote]);
+    if (totalSideStake == 0) { return (0, 0); }
+
+    uint256 stakeFraction = wdiv(stakes[_motionId][_staker][_vote], totalSideStake);
+    uint256 realStake = wmul(stakeFraction, motion.stakes[_vote]);
+
+    uint256 stakerReward;
+    uint256 repPenalty;
+
+    uint256 sumVotes;
+
+    for (uint256 i; i < motion.votes.length; i++) {
```

Comment:
Given we only care if `sumVotes` is greater than 0, this loop could be rewritten to be more efficient in most cases (i.e. once it's nonzero, we have no need to look at the rest of the votes).